### PR TITLE
RolePicker: UI improvements 

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
Minor UI improvements to the role picker: 
- Ensure the input and display widths are the same, to avoid jarring expanding when opened. 
- Decrease the width of input so it doesn't take too much space when closed. 
- Apply input styles to the selected roles display to clearly signal that it is editable.
- Fix a bug with menu position not updated on role change.

**Which issue(s) this PR fixes**:
Part of https://github.com/grafana/grafana-enterprise/issues/2992

**Special notes for your reviewer**:

https://user-images.githubusercontent.com/8878045/169810200-ef4a097c-2170-4639-a710-b4352c91292a.mov

### CODE REVIEW

**Comment by eleijonmarck** [comment]
AHhHHHH yeas ! @Clarity-89 a field around the input field is so nice.

File: public/app/core/components/RolePicker/ValueContainer.tsx [hunk 1]
```diff
@@ -8,15 +8,17 @@ export interface Props {
   children: ReactNode;
   iconName?: IconName;
 }
-export const ValueContainer = ({ children, iconName }: Props) => {
+export const ValueContainer = forwardRef<HTMLDivElement, Props>(({ children, iconName }, ref) => {
```
Comment (eleijonmarck) [inline public/app/core/components/RolePicker/ValueContainer.tsx:11]: @Clarity-89 what is a forwardRef? interesting

**Review by eleijonmarck — APPROVED** [review]
⚡

File: public/app/core/components/RolePicker/ValueContainer.tsx (hunk 1 above)
Comment (Clarity-89) [reply public/app/core/components/RolePicker/ValueContainer.tsx:11]: There's some good info in react docs: https://reactjs.org/docs/forwarding-refs.html

**Comment by VesnaDean** [comment]
I like this! :) One thing we should also do is make it possible to erase the assigned user roles using the back button on the keyboard and then clicking update. The interaction where you have to locate a role and uncheck the box feels cumbersome.

**Pull request merged**